### PR TITLE
Add halt feature for quartz scheduler

### PR DESCRIPTION
**Pull request by nragon**
## Proposed change
New quartz property for stopping scheduler from starting immediately.

## Description
A new property halt-start will be added to prevent scheduler from starting even if it finds `io.quarkus.scheduler.Scheduled` or force-start is set to true.
This will enable a better control on scheduler activity in cases where:
- Listeners need to be registered before scheduler start
- Scheduler instance is used as a orchestrator rather than a worker

### CODE REVIEW

File: extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java [hunk 1]
```diff
@@ -33,4 +33,12 @@
...
     public boolean forceStart;
 
+    /**
+     * Scheduler will immediately start running if it finds a {@link io.quarkus.scheduler.Scheduled} business method
+     * or is forced to start using forceStart property.
+     * If set to true the scheduler will be not start triggering jobs until an explicit start is called from the main
+     * scheduler. This is useful to programmatically register listeners before scheduler starts performing some work.
+     */
+    @ConfigItem
+    public boolean haltStart;
```
Comment (mkouba) [inline extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java:43]: I wonder if we should introduce a new config property, e.g. `startMode` with values `normal`, `forced` and `halted` and deprecate the `forceStart` instead. WDYT?

CC @machi1990

Comment (machi1990) [reply extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java:43]: +1 Great thinking.

**Comment by gsmet** [comment]
@nragon interested in pursuing @mkouba 's idea? We would need to make the existing one as deprecated and take it into account except if the mode defined in  `startMode` is not `normal`, in which case, the new config property would override it.

FYI, we release 1.11 CR1 on Wednesday morning so if you want this in 1.11, we need to merge it before tomorrow evening.

File: extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java (hunk 1 above)
Comment (nragon) [reply extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java:43]: LGTM

**Comment by nragon** [comment]
@gsmet Already making the changes

File: extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java [hunk 2]
```diff
@@ -29,8 +31,23 @@
...
+     * By default, the scheduler is not started unless a {@link io.quarkus.scheduler.Scheduled} business method
+     * is found. (normal mode)
+     * If set to "forced", scheduler will be started even if no scheduled business methods are found.
+     * This is necessary for "pure" programmatic scheduling.
+     * Additionally, setting it to "halted" will behave just like forced mode but the scheduler will be not start
+     * triggering jobs until an explicit start is called from the main scheduler.
+     * This is useful to programmatically register listeners before scheduler starts performing some work.
+     */
+    @ConfigItem
+    public Optional<String> startMode;
```
Comment (machi1990) [inline extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java:52]: Let's re-use the enum declared in https://github.com/quarkusio/quarkus/pull/14043/files#diff-4996139067b2db98f1af6397a79d30707d6fa1a376ba3afa453c95bcaa012582R6

File: extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzStartMode.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    FORCED("forced"),
+
+    /**
+     * Just like forced mode but the scheduler will be not start triggering jobs until an explicit start is called
+     * from the main scheduler.
+     * This is useful to programmatically register listeners before scheduler starts performing some work.
+     */
+    HALTED("halted");
+
+    private final String mode;
```
Comment (machi1990) [inline extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzStartMode.java:26]: This is not needed.

**Comment by machi1990** [comment]
> @gsmet Already making the changes

Thank you for applying the changes. I've requested two small changes other than that it LGTM.

**Comment by machi1990** [comment]
Thanks @nragon for applying the feedback quickly. Do you mind squashing the commits into one commit? Afterward it will be good to go.

**Comment by gsmet** [comment]
I will squash things and make a few minor adjustments.

**Comment by gsmet** [comment]
Hum, the history is completely borked :/ I have conflicts just squashing the commits.

**Comment by gsmet** [comment]
I fixed the history but given you created the PR from your master, I won't rewrite it. Will open a new PR.

**Comment by gsmet** [comment]
New PR: https://github.com/quarkusio/quarkus/pull/14122

**Pull request closed without merging**